domain/consensus/processes/pruningmanager/pruningmanager.go

```diff
@@ -237,28 +274,64 @@ func (pm *pruningManager) deleteBlock(blockHash *externalapi.DomainHash) (alread
 	return false, nil
 }
 
-func (pm *pruningManager) CalculatePruningPointByHeaderSelectedTip() (*externalapi.DomainHash, error) {
+func (pm *pruningManager) IsValidPruningPoint(block *externalapi.DomainHash) (bool, error) {
 	headersSelectedTip, err := pm.headerSelectedTipStore.HeadersSelectedTip(pm.databaseContext)
 	if err != nil {
-		return nil, err
+		return false, err
+	}
+
+	// A pruning point has to be in the selected chain of the headers selected tip.
+	headersSelectedTipGHOSTDAGData, err := pm.ghostdagDataStore.Get(pm.databaseContext, headersSelectedTip)
+	if err != nil {
+		return false, err
+	}
+
+	isInSelectedParentChainOfHeadersSelectedTip, err := pm.dagTopologyManager.IsInSelectedParentChainOf(block,
```

Comment:
is `pm.dagTopologyManager.IsInSelectedParentChainOf(a,a)==true`?

Reply:
Yes